rct229/rules/section21/section21rule10.py

```diff
@@ -0,0 +1,112 @@
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import HVAC_SYS
+from rct229.ruleset_functions.get_baseline_system_types import get_baseline_system_types
+from rct229.ruleset_functions.get_hw_loop_zone_list_w_area_dict import (
+    get_hw_loop_zone_list_w_area,
+)
+from rct229.schema.config import ureg
+from rct229.utils.assertions import getattr_
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_1,
+    HVAC_SYS.SYS_5,
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_11_2,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_1A,
+    HVAC_SYS.SYS_7A,
+    HVAC_SYS.SYS_11_2A,
+    HVAC_SYS.SYS_12A,
+    HVAC_SYS.SYS_1B,
+    HVAC_SYS.SYS_3B,
+    HVAC_SYS.SYS_5B,
+    HVAC_SYS.SYS_6B,
+    HVAC_SYS.SYS_7B,
+    HVAC_SYS.SYS_8B,
+    HVAC_SYS.SYS_9B,
+    HVAC_SYS.SYS_11_1B,
+    HVAC_SYS.SYS_12B,
+    HVAC_SYS.SYS_1C,
+    HVAC_SYS.SYS_3C,
+    HVAC_SYS.SYS_7C,
+    HVAC_SYS.SYS_11_1C,
+    HVAC_SYS.SYS_12C,
+]
+
+PUMP_SPEED_CONTROL = schema_enums["PumpSpeedControlOptions"]
+PUMP_CONFIGURATION_THRESHOLD = 12000 * ureg("ft2")
+
+
+class Section21Rule10(RuleDefinitionListIndexedBase):
+    """Rule 10 of ASHRAE 90.1-2019 Appendix G Section 23 (Hot water loop)"""
+
+    def __init__(self):
+        super(Section21Rule10, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section21Rule10.PumpRule(),
+            index_rmr="baseline",
+            id="21-10",
+            description="When the building is modeled with HHW plant (served by either boiler(s) or purchased hot "
+            "water/steam), the hot water pump shall be modeled as riding the pump curve if the hot water "
+            "system serves less than 120,000 ft^2 otherwise it shall be modeled with a VFD.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="pumps[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list contains all HVAC systems that are modeled in the rmi_b
+        available_type_lists = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+        return any(
+            [
+                available_type in APPLICABLE_SYS_TYPES
+                for available_type in available_type_lists
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        # to avoid pumps in service water heating system
+        loop_zone_list_w_area_dict = get_hw_loop_zone_list_w_area(rmi_b)
+
+        return {"loop_zone_list_w_area_dict": loop_zone_list_w_area_dict}
+
+    def list_filter(self, context_item, data):
+        pump = context_item.baseline
```

Comment:
It might be good to add a comment explaining that this method is filtering for pumps with HEATING loops, using the fact that the  keys in `loop_zone_list_w_area_dict` are ids to `HEATING` loops.